### PR TITLE
Fix test file discovery regression

### PR DESCRIPTION
**Pull request by vp-elitnet**
Proposed fix for #284.

### CODE REVIEW

**Comment by imcdo** [comment]
As many frameworks do (like for instance pytest) They filter files based on a prefixed or postfixed 'test' or something.  I would argue that this is ideal as it allows for not test files to not be collected around other test files, however i see your point that this could be a blocker if you wanted to change the pattern, so i would prefer making the prefix/postfix test file identifier be configurable as it is in something like pytest

**Comment by imcdo** [comment]
looks good, but before this gets merged in, I would like it if you could add documentation for this behavior as it could be a bit unclear if someone can run a test individually but it not be collected when specifying its parent directory https://github.com/confluentinc/ducktape/blob/master/docs/run_tests.rst

**Comment by stan-confluent** [comment]
Thanks for finding the issue and a suggested fix! Can you please also add a test to ensure that passing the folder with a mix of test and non-test files still behaves as expected (ie non-test files are ignored)?
Thanks!

**Comment by vp-elitnet** [comment]
Added behaviour clarification in "Running Tests" section.

> Can you please also add a test to ensure that passing the folder with a mix of test and non-test files still behaves as expected (ie non-test files are ignored)?

Added a minimal additional mock test module to verify this case.

**Review by imcdo — APPROVED** [review]
Looks good thanks so much!

**Review by stan-confluent — APPROVED** [review]
Thank you!

File: tests/loader/resources/loader_test_directory/name_does_not_match_pattern.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from ducktape.tests.test import Test
+
```
Comment (imcdo) [inline tests/loader/resources/loader_test_directory/name_does_not_match_pattern.py:2]: Nit: styling required for the pr job:
```suggestion


```

**Pull request merged**